fdbserver/DDTeamCollection.actor.cpp

```diff
@@ -3458,6 +3516,64 @@ Future<Void> DDTeamCollection::serverTeamRemover() {
 	return DDTeamCollectionImpl::serverTeamRemover(this);
 }
 
+Future<Void> DDTeamCollection::fixWrongReplicasLoop() {
+	return DDTeamCollectionImpl::fixWrongReplicasLoop(this);
+}
+
+void DDTeamCollection::fixWrongReplicas() {
+	int maxTeamSize = maxLargeTeamSize();
+	int checkCount = 0;
+	for (auto& it : underReplication.ranges()) {
+		if (!it.value()) {
+			// The key range is not under-replicated
+			continue;
+		}
+		for (auto& r : shardsAffectedByTeamFailure->intersectingRanges(it.range())) {
+			if (++checkCount > SERVER_KNOBS->DD_MAXIMUM_LARGE_TEAM_CLEANUP) {
+				return;
+			}
+
+			auto& teams = r.value();
+			if (!teams.second.empty()) {
+				// The key range is currently being moved
+				continue;
+			}
+
+			int customReplicas = configuration.storageTeamSize;
+			for (auto& c : customReplication->intersectingRanges(r.range())) {
+				customReplicas = std::max(customReplicas, c.value());
+			}
```

Comment:
Is this `intersectingRanges` logic still needed after you've added the splitting for ranges?

Reply:
it is more robust if this code still properly handles the case where a single shard has multiple custom replicas. This will not cost us anything since we will generally only have one intersecting range